### PR TITLE
feature: allow using your own COMMIT_EDITMSG syntax

### PR DESCRIPTION
**Pull request by yyyc514**

### CODE REVIEW

**Comment by yyyc514** [comment]
Ok, I've squash this down to JUST this change without anything else.

File: core/commands/commit.py [hunk 1]
```diff
@@ -44,7 +45,14 @@ def run_async(self, repo_path=None, include_unstaged=False, amend=False):
...
         view.settings().set("git_savvy.repo_path", repo_path)
-        view.set_syntax_file("Packages/GitSavvy/syntax/make_commit.tmLanguage")
+
+        gitsavvy_settings = sublime.load_settings("GitSavvy.sublime-settings")
+        if gitsavvy_settings.get("use_syntax_for_commit_editmsg")
+            syntax_file = util.file.get_syntax_for_file("COMMIT_EDITMSG")
+            view.set_syntax_file(syntax_file)
+        else:
+            view.set_syntax_file("Packages/GitSavvy/syntax/make_commit.tmLanguage")
+
```
Comment (divmain) [inline core/commands/commit.py]: Overriding the default syntax is a good approach to resolve #41.  However, let's simplify by creating a setting in `GitSavvy.sublime-settings` called `commit_syntax` with a default value of `Packages/GitSavvy/syntax/make_commit.tmLanguage` (and remove `use_syntax_for_commit_editmsg`).  Then, replace the above chunk of code with:
```python
gitsavvy_settings = sublime.load_settings("GitSavvy.sublime-settigns")
view.set_syntax_file(gitsavvy_settings.get("commit_syntax")
```

**Comment by yyyc514** [comment]
I felt this was a lot simper for veteran an novice user alike - *and makes use of the existing setup that Sublime already provides for choosing which template you want for particular files.*  We don't need to re-invent the wheel here,  Sublime already has the feature to let pros decide exactly which syntax applies to exactly which files.  We just need to make it clear if GitSavvy is in charge or some other plugin the user has decided to install and configure.

Honestly I'm not even sure how to easily get the full path of a syntax file inside a plugin I just installed without opening up the settings dir and looking manually and typing it out.

**Comment by yyyc514** [comment]
If there is a clearer name I'd be for that, but I think a True/False toggle is more in the spirit of Sublime and keeping things truly simple.

**Comment by yyyc514** [comment]
I'm not even so sure the default behavior should be:

- check for COMMIT_EDITMSG syntax
- if found, use that
- if not, use our built-in

But then I prefer guessing in the 90% cases and saving configuration for the remaining 10.

**Comment by divmain** [comment]
I'll consider this.  Can you point me to other packages that take a similar approach?  I'm hesitant to go this direction because the commit view is an interactive view, not a file type.  That being the case, assigning syntax by a fake file extension seems like an opaque approach to me.

**Comment by yyyc514** [comment]
> That being the case, assigning syntax by a fake file extension seems like an opaque approach to me.

I don't see it that way at all.  That's what git calls it's PRE-COMMIT edit file... so it just seemed the simplest thing to allow someone to reuse a theme that already targeted that file.  It really comes down to what the view is and does.  I'd even suggest changing the name of the view to COMMIT_EDITMSG, but I didn't think you'd go fo that.

Right now it's really a glorified COMMIT_EDITMSG.  It might do more in the future, but that's how I'm thinking of it.  Obviously if it starts doing really wacky things that aren't compatible with the COMMIT_EDITMSG syntax then this would be a moot argument - but if what it does it a million things WITHIN this format then it still seems maybe a good idea to allow an easy toggle.

What about two settings?

- commit_syntax
- use_syntax_for_commit_editmsg

Or would you allow special meanings of some values?

- commit_syntax: "" / "default" (internal syntax)
- commit_syntax: "COMMIT_EDITMSG"
- commit_syntax: "/your/favorite/file.tmLanguage"

**Comment by yyyc514** [comment]
Of course the settings would be documented in the sample config file.

**Comment by yyyc514** [comment]
> Can you point me to other packages that take a similar approach?

I have no idea.  My point is that it's the approach of Sublime itself - to color files based on what they are - determined by extension.  This is a pre-commit git message view.  The fact that we allow adding commit summary, diff, etc. to that view (just like git itself) make that even more clear.  So until it differs substantially from that type of view it seemed to make sense to "color files based on what they are", as I put it.

>This is a pre-commit git message view.

I'm not sure that changes as it gets more powerful.  I'd actually love to be able to stage files right from the summary view on the commit page... but if I could that wouldn't change me thinking of the view as a pre-commit git message view.

Obviously those using a less-specific syntax might have fewer coloring feature (if you start adding keystroke colors, etc), but that would be their choice.  I'm not suggesting make this default, only an option.

**Comment by yyyc514** [comment]
> commit_syntax: "COMMIT_EDITMSG"

My chief objection is not the name of the setting but the user having to waste time figuring out the exact path name of a syntax file - when they just want the editor to have the same behavior as it does for other "file types".

**Comment by divmain** [comment]
Alright, I'm not really a fan of treating the filename like a filename extension, but I've decided its not terribly important to me, seeing as this feature is disabled by default.  So I'm going to merge this in.

Keep in mind, however, that the implementation and/or behavior may change once the commit view is converted to being more interface-like, due to constraints for Sublime's region-based syntax highlighting.  I'll do my best to preserve the functionality.  For that reason, I'm not going to include the change in a changelog until I'm sure the feature won't go away or significantly change.

**Comment by yyyc514** [comment]
> Keep in mind, however, that the implementation and/or behavior may change once the commit view is converted to being more interface-like, due to constraints of Sublime's region-based multiple-syntax highlighting. I'll do my best to preserve the functionality.

At the very least then GitSavvy should (please) have an option to use a toggle to use an old "git style" commit screen (and related syntax) or the "new, amazing, and so completely unlike git commit that a syntax file can't work" dashboard/commit view.  We're all waiting on the edge of our seats to see what you want to do with git commit but honestly it's one of the few screens I haven't yet been hankering for any improvements too - beyond supporting what git already does from the console.

Maybe it's because I just don't know what is possible. :-)  I wanted to stage files from there but I think the new improvements to quick stage may have done the trick to scratch that itch.

**Comment by yyyc514** [comment]
> Alright, I'm not really a fan of treating the filename like a filename extension, but I've decided its not terribly important to me, seeing as this feature is disabled by default.

Thanks. :-)  I'll see what I can do to move the Pedantic stuff forward now that this is in.

**Pull request merged**